### PR TITLE
Refactor/1839/refactor test for category item component1

### PR DESCRIPTION
**Pull request by Pipuss**
## Description

Refactored test for category-item component by RTL

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/pages/home/categories-list/category-item/tests/category-item.spec.js [hunk 1]
```diff
@@ -13,12 +15,14 @@ const mockDispatch = jest.fn();
...
-  it('Should render CategoryItem', () => {
-    expect(wrapper).toBeDefined();
+  const { container } = render(
+    <BrowserRouter>
+      <CategoryItem categoryName='testName' categoryImageUrl='testImageUrl' categoryUrl='testUrl' />
+    </BrowserRouter>
+  );
+
+  test('Should render CategoryItem', () => {
+    expect(container).toBeInTheDocument();
```
Comment (ElitKondor) [inline src/pages/home/categories-list/category-item/tests/category-item.spec.js:26]: Can we restructure the test by doing a test on something more specific like a button, input field or text content?

Comment (Pipuss) [reply src/pages/home/categories-list/category-item/tests/category-item.spec.js:26]: Yes, i will try something

File: src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  );
+
+  test('Should render correct structure of CategoryItem component', () => {
+    const linkElement = container.querySelector('div a');
+    expect(linkElement.getAttribute('href')).toBe('/testUrl');
+
+    const categoryElement = container.querySelector('div a span');
+    expect(categoryElement.textContent).toBe('testName');
+
+    const svgElement = container.querySelector('div a div svg');
```
Comment (afanasieva-anastasiia) [inline src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js:32]: Сan we search for an element with some text, testid, etc?

**Review by afanasieva-anastasiia — CHANGES_REQUESTED** [review]

File: src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js (hunk 2 above)
Comment (Pipuss) [reply src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js]: Solved :-)

**Review by maarekua — APPROVED** [review]

File: src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const { getByRole, getAllByText } = render(
+    <BrowserRouter>
+      <CategoryItem {...mockProps} />
+    </BrowserRouter>
+  );
+
+  test('Should render CategoryItem component', () => {
+    const linkElement = getByRole('link');
+    expect(linkElement.getAttribute('href')).toBe('/testUrl');
+
```
Comment (SirAndrii) [inline src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js:31]: What would you do if there is not only one link? Maybe try use {name: 'anchor'} ?

Comment (Pipuss) [reply src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js:31]: Good remark.
Can you give a hint for beginner? 
What exactly can I do with links? 
Can I write a test-id attribute in the component?

Comment (SirAndrii) [reply src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js:31]: ok, that also could be solved with getByRole('link',{name: 'acncor_text_of_the_link'})

Comment (Pipuss) [reply src/tests/unit/pages/home/categories-list/category-item/category-item.spec.js:31]: Ok, thanks. 
I will learn such a way

**Review by SirAndrii — APPROVED** [review]

**Review by afanasieva-anastasiia — APPROVED** [review]

**Pull request merged**